src/util.rs

```diff
@@ -80,61 +81,53 @@ impl<P: AsRef<Path> + Send + Sync> FileIntoMods<P> {
                 });
         }
 
-        let mut sub_self = self.sub_mod(crate_name.as_ref());
-        let folded_crate = noop_fold_crate(&mut sub_self, rust_crate);
+        let mut sub_self = self.sub_mod(&mod_name);
+        let folded_mod = fold_item_mod(&mut sub_self, item_mod);
         let file_name = self
             .current_dir
             .as_ref()
-            .join(String::from(crate_name.as_ref()) + ".rs");
+            .join(mod_name.to_owned() + ".rs");
         trace!(
             "Writing contents of module {} to file {}",
-            crate_name,
+            mod_name,
             file_name.display()
         );
-        write_crate(folded_crate, &file_name)
+        write_mod(folded_mod, &file_name)
             .unwrap_or_else(|err| panic!("writing to {} failed with {}", file_name.display(), err));
         Ok(())
     }
 }
 
-impl<P: AsRef<Path> + Send + Sync> Folder for FileIntoMods<P> {
+impl<P: AsRef<Path> + Send + Sync> Fold for FileIntoMods<P> {
     fn fold_item(&mut self, mut item: Item) -> Item {
-        for rust_crate in extract_crate_from_mod(&mut item.node) {
-            self.fold_sub_crate(&item.ident, rust_crate).unwrap();
+        for item_mod in extract_mod(&mut item) {
+            self.fold_sub_mod(item_mod).unwrap();
         }
-        noop_fold_item(self, item)
+        fold_item(self, item)
     }
 }
 
-fn write_crate(rust_crate: Crate, file_name: &Path) -> Result<(), Error> {
+fn write_mod(item_mod: ItemMod, file_name: &Path) -> Result<(), Error> {
     trace!("Opening file {}", file_name.display());
     let mut file = File::create(&file_name)
         .context(format_err!("unable to create file {}", file_name.display()))?;
     trace!("Successfully opened file {}", file_name.display());
     debug!("Writing to file {}", file_name.display());
-    write_all_tokens(&rust_crate, &mut file)
+    write_all_tokens(&item_mod, &mut file)
 }
 
-fn extract_crate_from_mod<'a>(node: &'a mut ItemKind) -> Option<Crate> {
-    if let ItemKind::Mod(ref mut maybe_items) = *node {
-        maybe_items.take().map(make_crate)
+fn extract_mod<'a>(node: &'a mut Item) -> Option<ItemMod> {
+    if let Item::Mod(mod_item) = &*node {
+        Some(mod_item.clone())
```

Comment:
I'm pretty sure that this is the problem line - when you fold over a `mod foo { ... }` you need to convert it into `mod foo;`  which the `maybe_items.take()` was doing.

Reply:
Yes, you are right. Sorry again.